src/BotPlutusInterface/Contract.hs

```diff
@@ -71,8 +81,20 @@ runContract ::
   ContractEnvironment w ->
   Contract w s e a ->
   IO (Either e a)
-runContract contractEnv (Contract effs) = do
-  runM $ handlePABEffect @w contractEnv $ raiseEnd $ handleContract contractEnv effs
+runContract contractEnv contract =
```

Comment:
It might not be too much of a performance gain, but I think we could avoid accumulating TxStats when it is not needed.
- Leaving this function as it was, without wrapping
- We could introduce some flag, or inject the `saveBudgetImpl` handler through `handlePABEffect`, with an ignore option